src/averageAges.js

```diff
@@ -56,8 +67,27 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+
+  const getChildren = people.filter(child => onlyWithSon
+    ? people.find(person => person.name === child.mother) && child.sex === 'm'
+    : people.find(person => person.name === child.mother)
+  );
+
+  const ageDifference = getChildren.map(child => {
+    const mother = people.find(person => person.name === child.mother);
+
+    return child.born - mother.born;
+  });
+
+  return ageDifference.reduce((a, b) => a + b, 0) / ageDifference.length;
 }
 
+const calculateAverageHelper = (array) => {
+  const len = array.length;
+
+  return array.map(a => a.died - a.born).reduce((x, y) => x + y, 0) / len;
```

Comment:
For better readability. 
 ```suggestion
   return array
     .map(person => a.died - a.born)
     .reduce((x, y) => x + y, 0) / len;
```
You can even change the 'a' with 'person'. But I think it is fine.